include/render.h

```diff
@@ -94,17 +96,70 @@ struct Render_t {
 	bool fullFrame = true;
 };
 
+struct RenderedImage {
+	uint16_t width              = 0;
+	uint16_t height             = 0;
+	bool double_width           = false;
+	bool double_height          = false;
+	bool flip_vertical          = false;
+	Fraction pixel_aspect_ratio = {};
+	uint8_t bits_per_pixel      = 0;
+	uint16_t pitch              = 0;
+	uint8_t* image_data         = nullptr;
+	uint8_t* palette_data       = nullptr;
+
+	inline bool is_paletted() const
+	{
+		return (bits_per_pixel == 8);
+	}
+
+	RenderedImage deep_copy() const
+	{
+		RenderedImage copy = *this;
+
+		// Deep-copy image and palette data
+		const auto image_data_num_bytes = height * pitch;
+
+		copy.image_data = new uint8_t[image_data_num_bytes];
+
+		assert(image_data);
+		std::memcpy(const_cast<uint8_t*>(copy.image_data),
+		            image_data,
+		            image_data_num_bytes);
+
+		// TODO it's bad that we need to make this assumption downstream
+		// on the size and alignment of the palette...
+		if (palette_data) {
+			constexpr auto PaletteNumBytes = 256 * 4;
+			copy.palette_data = new uint8_t[PaletteNumBytes];
+
+			std::memcpy(const_cast<uint8_t*>(copy.palette_data),
```

Comment:
I think the two `const_cast<uint8_t*>` can be eliminated, because these variables are already non-const `uint8_t*` variables (so should be able to pass them in straight-away).

Reply:
I had to do these to make all compilers happy on all platforms, after much trial and error.

I can try removing it again, but if I get CI warnings because of warnings, back it goes...